oshi-core/src/main/java/oshi/hardware/common/AbstractPowerSource.java

```diff
@@ -46,11 +55,27 @@
      * @param newTimeRemaining
      *            Seconds of time remaining
      */
+    @Deprecated

```

Comment:
Good effort but not necessary. The OSHI API is documented as being the interfaces and classes in the `oshi.hardware` and `oshi.software.os` packages.  Constructors in the class-specific (or abstract) classes aren't API and it's okay to remove this one (and the class-specific variants) without documentation.